lib/internal/test_runner/mock/mock.js

```diff
@@ -61,9 +61,14 @@ class MockFunctionContext {
   mockImplementationOnce(implementation, onCall) {
     validateFunction(implementation, 'implementation');
     const nextCall = this.#calls.length;
-    const call = onCall ?? nextCall;
-    validateInteger(call, 'onCall', nextCall);
-    this.#mocks.set(call, implementation);
+    if (onCall) {
+      validateInteger(onCall, 'onCall', nextCall);
+      this.#mocks.set(onCall, implementation);
+    } else {
+      this.#mocks.set(this.#mocks.size + nextCall, implementation);
```

Comment:
I don't think this will work:

```js
  let cnt = 0;
  function addOne() {
    cnt++;
    return cnt;
  }

  const fn = t.mock.fn(addOne);
  fn.mock.mockImplementationOnce(() => 100, 100);
  fn.mock.mockImplementationOnce(() => 200, 200);
  fn.mock.mockImplementationOnce(() => 300, 300);
  fn.mock
    .mockImplementationOnce(() => 999)
    .mockImplementationOnce(() => 888);

  console.log(fn());
  console.log(fn());
  console.log(fn());
  console.log(fn());
  console.log(fn());
```

The expected output here is 999, 888, 1, 2, 3. However, the actual output is 1, 2, 3, 999, 888.

Reply:
fixed